pkg/docker/docker.go

```diff
@@ -241,6 +241,93 @@ func (c *Client) SupportedPlatforms(ctx context.Context) ([]v1.Platform, error)
 	}, nil
 }
 
+// ImageDistribution fetches the details for the specified image by asking
+// docker to fetch the distribution manifest from the registry. The digest
+// retrieved may not appear accurate when compared to the hub website but
+// this is expected.
+//
+// cf:
+//   - https://github.com/moby/moby/issues/40636)
+//   - https://github.com/docker/roadmap/issues/262
+//
+// When a docker image is available on only a single platform, the digest
+// shown will be the digest pointing directly at the manifest for that image
+// on that platform (as shown by the docker hub).  Where multiple platforms
+// are available, the digest is pointing to a top level document describing
+// all of the different platform manifests available.
+//
+// In either case, `docker pull` will do the correct thing and download the
+// image for your platform. For example:
+//
+// $ docker manifest inspect  bitnami/rabbitmq@sha256:0be0d2a2 ...
+//
+//	"manifests": [ {
+//		  "digest": "sha256:959a02013e8ab5538167f9....",
+//		  "platform": { "architecture": "amd64", "os": "linux" }
+//		},
+//		{
+//		  "digest": "sha256:11ee2c7e9e69e3a8311a19....",
+//		  "platform": { "architecture": "arm64", "os": "linux"}
+//		}]
+//
+// $ docker pull bitnami/rabbitmq@sha256:0be0d2a2 ...
+// $ docker image ls
+// bitnami/rabbitmq ... 48603925e10c
+//
+// The digest 486039 can be found in manifest sha256:11ee2c7e which is the manifest for
+// the current authors machine.
+//
+// $ docker manifest inspect bitnami/rabbitmq@sha256:11ee2c7e
+//
+//	  "config": {
+//		   "size": 7383,
+//		   "digest": "sha256:48603925e10c01936ea4258f...."
+//	  }
+//
+// This is the image that will finally be installed.
+func (c *Client) ImageDistribution(
+	ctx context.Context, image string,
+	forceRemote bool, creds config.DockerCredentials,
+) (*ImageManifest, error) {
+	// Check the local repository first to see if we already have this image and if so,
+	// return those details, but only if the caller did not set forceRemote
+	if !forceRemote {
+		info, _, err := c.ImageInspectWithRaw(ctx, image)
+		if err == nil {
+			repos := info.RepoDigests
+			if len(repos) >= 1 {
+				// We only want the digest part of the name, otherwise we would have
+				// to go through supporting two different values in the returned
+				// ImageManifest (fully qualified IDs and also just digests)
+				digestParts := strings.Split(repos[0], ":")
```

Comment:
Would it be safer to use `digest.Digest(repos[0]).Encoded()` here?

Or even, shouldn't the `digest` field be a full `Digest` including the algorithm?

Reply:
Have switched to using digest.Part, but each repo here is the full image name with a digest - so we still don't want the image name.